tick/stateful/evaluation_funcs.go

```diff
@@ -0,0 +1,789 @@
+package stateful
+
+import (
+	"regexp"
+
+	"github.com/influxdata/kapacitor/tick"
+)
+
+type operationKey struct {
+	operator  tick.TokenType
+	leftType  ValueType
+	rightType ValueType
+}
+
+var boolTrueResultContainer = resultContainer{BoolValue: true, IsBoolValue: true}
+var boolFalseResultContainer = resultContainer{BoolValue: false, IsBoolValue: true}
+var emptyResultContainer = resultContainer{}
+
+var evaluationFuncs = map[operationKey]evaluationFn{
+	// -----------------------------------------
+	//	Comparison evaluation funcs
+
+	operationKey{operator: tick.TokenAnd, leftType: TBool, rightType: TBool}: func(scope *tick.Scope, executionState ExecutionState, leftNode, rightNode NodeEvaluator) (resultContainer, *ErrSide) {
+		var left bool
+		var right bool
+		var err error
+
+		if left, err = leftNode.EvalBool(scope, executionState); err != nil {
+			return boolFalseResultContainer, &ErrSide{error: err, IsLeftSide: true}
+		}
+
+		// Short circuit evaluation
```

Comment:
>If this is breaking change - we can remove the short circuit evaluation.

I think most people will expect it short circuit the evaluation since most languages do. Although in combination with stateful expressions it may be a bad idea. What do you think?